### ISSUE TITLE
Apache Solr integration documentation outdated

### ISSUE THREAD

**Issue by SalahAdDin** [django-oscar/django-oscar#2096]
Hi guys

As[ i couldn't integrate](https://github.com/django-oscar/django-oscar/issues/2089) Oscar with `elasticsearch` search engine, i had chosen use `apache solr` instead.

```
 I test it in the same way that i test `elasticsearch`: seeking in the home page all products, and i have this bug:
[13/Aug/2016 01:43:49] "GET / HTTP/1.1" 200 28031
Failed to query Solr using '*:*': Solr responded with an error (HTTP 404): [Reason: Error 404 Not Found]
Traceback (most recent call last):
  File "/home/salahaddin/Proyectos/demo-oscar/lib/python3.5/site-packages/haystack/backends/solr_backend.py", line 134, in search
    raw_results = self.conn.search(query_string, **search_kwargs)
  File "/home/salahaddin/Proyectos/demo-oscar/lib/python3.5/site-packages/pysolr.py", line 710, in search
    response = self._select(params, handler=search_handler)
  File "/home/salahaddin/Proyectos/demo-oscar/lib/python3.5/site-packages/pysolr.py", line 411, in _select
    return self._send_request('get', path)
  File "/home/salahaddin/Proyectos/demo-oscar/lib/python3.5/site-packages/pysolr.py", line 386, in _send_request
    raise SolrError(error_message % (resp.status_code, solr_message))
pysolr.SolrError: Solr responded with an error (HTTP 404): [Reason: Error 404 Not Found]
Failed to query Solr using '*:*': Solr responded with an error (HTTP 404): [Reason: Error 404 Not Found]
Traceback (most recent call last):
  File "/home/salahaddin/Proyectos/demo-oscar/lib/python3.5/site-packages/haystack/backends/solr_backend.py", line 134, in search
    raw_results = self.conn.search(query_string, **search_kwargs)
  File "/home/salahaddin/Proyectos/demo-oscar/lib/python3.5/site-packages/pysolr.py", line 710, in search
    response = self._select(params, handler=search_handler)
  File "/home/salahaddin/Proyectos/demo-oscar/lib/python3.5/site-packages/pysolr.py", line 411, in _select
    return self._send_request('get', path)
  File "/home/salahaddin/Proyectos/demo-oscar/lib/python3.5/site-packages/pysolr.py", line 386, in _send_request
    raise SolrError(error_message % (resp.status_code, solr_message))
pysolr.SolrError: Solr responded with an error (HTTP 404): [Reason: Error 404 Not Found]
```

So, i went to [integration docs](http://django-oscar.readthedocs.io/en/latest/howto/how_to_setup_solr.html) for know how i can integrate Oscar with `solr`. 
I had installed `solr` and i running it, i test this in browser with the url `http://localhost:8983/solr`, is `6.1.0` version.

Then i'd putted the concection setting:

``` python
HAYSTACK_CONNECTIONS = {
    'default': {
        'ENGINE': 'haystack.backends.solr_backend.SolrEngine',
        'URL': 'http://127.0.0.1:8983/solr',
        'INCLUDE_SPELLING': True,
        # 'INDEX_NAME': 'rosa-blanca',
    },
}
```

And then i tried to build the index, as step say `./manage.py rebuild_index --noinput`, but i get this error:

``` bash
 ./manage.py rebuild_index                                                                                         ✓  2031  00:47:57 
/home/salahaddin/Proyectos/demo-oscar/lib/python3.5/site-packages/django/utils/six.py:808: RemovedInDjango110Warning: SubfieldBase has been deprecated. Use Field.from_db_value instead.
  return meta(name, bases, d)

/home/salahaddin/Proyectos/demo-oscar/lib/python3.5/site-packages/treebeard/admin.py:62: RemovedInDjango110Warning: Support for string view arguments to url() is deprecated and will be removed in Django 1.10 (got django.views.i18n.javascript_catalog). Pass the callable instead.
  {'packages': ('treebeard',)}),

/home/salahaddin/Proyectos/demo-oscar/lib/python3.5/site-packages/treebeard/admin.py:62: RemovedInDjango110Warning: django.conf.urls.patterns() is deprecated and will be removed in Django 1.10. Update your urlpatterns to be a list of django.conf.urls.url() instances instead.
  {'packages': ('treebeard',)}),

WARNING: This will irreparably remove EVERYTHING from your search index in connection 'default'.
Your choices after this are to restore from backups or rebuild via the `rebuild_index` command.
Are you sure you wish to continue? [y/N] y
Removing all documents from your index because you said so.
Failed to clear Solr index: Solr responded with an error (HTTP 404): [Reason: Error 404 Not Found]
Traceback (most recent call last):
  File "/home/salahaddin/Proyectos/demo-oscar/lib/python3.5/site-packages/haystack/backends/solr_backend.py", line 101, in clear
    self.conn.delete(q='*:*', commit=commit)
  File "/home/salahaddin/Proyectos/demo-oscar/lib/python3.5/site-packages/pysolr.py", line 906, in delete
    return self._update(m, commit=commit, waitFlush=waitFlush, waitSearcher=waitSearcher, handler=handler)
  File "/home/salahaddin/Proyectos/demo-oscar/lib/python3.5/site-packages/pysolr.py", line 471, in _update
    return self._send_request('post', path, message, {'Content-type': 'text/xml; charset=utf-8'})
  File "/home/salahaddin/Proyectos/demo-oscar/lib/python3.5/site-packages/pysolr.py", line 386, in _send_request
    raise SolrError(error_message % (resp.status_code, solr_message))
pysolr.SolrError: Solr responded with an error (HTTP 404): [Reason: Error 404 Not Found]
All documents removed.
Indexing 1 Products
Failed to add documents to Solr: Solr responded with an error (HTTP 404): [Reason: Error 404 Not Found]
Traceback (most recent call last):
  File "/home/salahaddin/Proyectos/demo-oscar/lib/python3.5/site-packages/haystack/backends/solr_backend.py", line 72, in update
    self.conn.add(docs, commit=commit, boost=index.get_field_weights())
  File "/home/salahaddin/Proyectos/demo-oscar/lib/python3.5/site-packages/pysolr.py", line 875, in add
    overwrite=overwrite, handler=handler)
  File "/home/salahaddin/Proyectos/demo-oscar/lib/python3.5/site-packages/pysolr.py", line 471, in _update
    return self._send_request('post', path, message, {'Content-type': 'text/xml; charset=utf-8'})
  File "/home/salahaddin/Proyectos/demo-oscar/lib/python3.5/site-packages/pysolr.py", line 386, in _send_request
    raise SolrError(error_message % (resp.status_code, solr_message))
pysolr.SolrError: Solr responded with an error (HTTP 404): [Reason: Error 404 Not Found]
```

So, seeking the documentation i can see that something is wrong:

> $ cd solr-4.7.2/example/solr/collection1
> $ mv conf conf.original
> $ ln -s <your_oscar_checkout>/sites/sandbox/deploy/solr conf

Here are at least two error:
- The first and the worst, in the current Oscar version, in the `sites` folder doesn't exist anymore the `deploy` folder. But, in the Oscar's demo i can find these folder, and the sorl folder, but i can't find the `conf` file.
  I haven't clear what have i do in this case, copy all folder? copy only the schema file? What?
- In the curren `solr` version doesn't exist one `solr` folder inside `examples` folder.
  `bash
  ~/solr-6.1.0  ls example/                                                                                                                     ✓  2026  01:55:59 
  cloud  example-DIH  exampledocs  files  films  README.txt  resources
  `

So, i don't know what i have to do here for integrate successfully Oscar with `Apache solr`.

Thank you!

**Comment by oneb** [comment]
I came here to raise an issue about the same problem. 

Looks like the `sites/sandbox/deeploy/solr` folder was removed by commit f62e59ac63464f0d9e33f4070b3492f3b30d07f9.

**Comment by SalahAdDin** [comment]
@oneb , yes, these is the first problem.

**Comment by sasha0** [comment]
Fixed by 1a30d7b9a572fbe507fc3f4ff4209693ae61b57e.
